crates/rust-analyzer/src/config.rs

```diff
@@ -49,6 +49,8 @@ config_data! {
         /// Run build scripts (`build.rs`) for more precise code analysis.
         cargo_runBuildScripts |
         cargo_loadOutDirsFromCheck: bool = "false",
+        /// Disable running build scripts (`build.rs`) for the `rustc_private` crates in `rustcSource`.
+        cargo_disableRustcBuildScripts: bool = "false",
```

Comment:
Let's move this and `rustcSource` into `rustc_private` section, so that we have

```
"rustcPrivate.src": ...,
"rustcPrivate.runBuildScripts": ...,
```

Reply:
Would `cargo.rustcPrivate.runBuildScripts` and `cargo.rustcPrivate.src` make sense? Since we only have special behaviour for the rustcSource for cargo, afaik.

Reply:
I'd rathe keep it top level. The fact that it is `rustcPrivate` is more important than that it is `cargo`.

Reply:
but no strong opinion here!